nixos/modules/services/web-servers/phpfpm/default.nix

```diff
@@ -241,21 +273,20 @@ in {
       daemonize = false;
     };
 
-    systemd.slices.phpfpm = {
-      description = "PHP FastCGI Process manager pools slice";
-    };
-
-    systemd.targets.phpfpm = {
-      description = "PHP FastCGI Process manager pools target";
-      wantedBy = [ "multi-user.target" ];
-    };
+    systemd.sockets = mapAttrs' (pool: poolOpts:
+      nameValuePair "phpfpm-${pool}" {
+        description = "PHP FastCGI Process manager socket for pool ${pool}";
+        wantedBy = [ "sockets.target" ];
+        before = [ "multi-user.target" ];
+        listenStreams = [ poolOpts.socket ];
+      }
+    ) cfg.pools;
 
     systemd.services = mapAttrs' (pool: poolOpts:
       nameValuePair "phpfpm-${pool}" {
         description = "PHP FastCGI Process Manager service for pool ${pool}";
-        after = [ "network.target" ];
-        wantedBy = [ "phpfpm.target" ];
-        partOf = [ "phpfpm.target" ];
+        # https://freedesktop.org/wiki/Software/systemd/DaemonSocketActivation/
+        environment.FPM_SOCKETS = "${poolOpts.socket}=3";
```

Comment:
Is the file descriptor no. 3 really correct in this case? Documentation is really sparse on that one…

Reply:
Given things aren't working, I still have a few issues to resolve, and as you mentioned the documentation is somewhat lacking... I can't confidently say so. More on this when I (eventually) get back to this PR. Help appreciated.

Reply:
@florianjacob I'm really not sure what to say. There is hardly any documentation on this, so all I can do is go off that. I have found a number of examples on the internet that agree with this documentation, so at least I know other people are doing it too :man_shrugging: